### PR TITLE
test: add more tests for Write and WriteAuthorizationModel API

### PR DESCRIPTION
**Pull request by miparnisari**
Quick follow-up of https://github.com/openfga/openfga/pull/1208

Suggestion to reviewers:

image

### CODE REVIEW

File: pkg/server/test/write_authzmodel.go [hunk 1]
```diff
@@ -467,7 +468,74 @@ type other
 
 		// conditions
 		{
-			name: "condition_valid",
+			name:             "do_not_allow_conditions_if_bool_is_false_and_conditions_received",
```
Comment (jpadilla) [inline pkg/server/test/write_authzmodel.go:471]:
```suggestion
			name:             "disabled_conditions_and_conditions_received",
```

File: pkg/server/test/write_authzmodel.go [hunk 2]
```diff
@@ -467,7 +468,74 @@ type other
...
+						Name:       "condition1",
+						Expression: "param1 == 'ok'",
+						Parameters: nil,
+					},
+				},
+			},
+			errCode: codes.InvalidArgument,
+		},
+		{
+			name:             "do_not_allow_conditions_if_bool_is_false_and_relation_references_condition",
```
Comment (jpadilla) [inline pkg/server/test/write_authzmodel.go:506]:
```suggestion
			name:             "disabled_conditions_and_relation_references_condition",
```

**Review by jpadilla — CHANGES_REQUESTED** [review]

**Review by jon-whit — CHANGES_REQUESTED** [review]
Left one minor comment - looks good otherwise.

**Review by jpadilla — APPROVED** [review]

File: pkg/server/test/write.go [hunk 3]
```diff
@@ -1334,20 +1431,24 @@ func TestWriteCommand(t *testing.T, datastore storage.OpenFGADatastore) {
...
 			}
 			resp, gotErr := cmd.Execute(ctx, test.request)
 
-			require.ErrorIs(gotErr, test.err)
+			if test.err != nil {
+				require.ErrorIs(t, gotErr, test.err)
+				require.ErrorContains(t, gotErr, test.err.Error())
+			}
 
 			if test.response != nil {
```
Comment (jon-whit) [inline pkg/server/test/write.go:1450]: By this point I assume `gotErr` is nil. Just to make sure we cover our bases, will you please make sure this is the case by adding an assertion for this case where `require.Nil(t, gotErr)`.

**Review by jon-whit — APPROVED** [review]

**Pull request merged**